src/stan/mcmc/hmc/nuts/base_nuts.hpp

```diff
@@ -249,8 +258,9 @@ namespace stan {
             z_propose = z_propose_right;
         }
 
+        Eigen::VectorXd rho_subtree = rho_left + rho_right;
         rho += rho_subtree;
```

Comment:
This should be

```
rho += rho_left + rho_right;
```

to minimize copies.  The assignment to `rho_subtree` would otherwise alloc and assign, whereas the direct `+=` will avoid it.